### DISCUSSION TITLE
IC8 Step Motor

### DISCUSSION THREAD

**Discussion by Mightydob** [doudar/SmartSpin2k#308] [Q&A] [answered]
Hi all, excited to find out about this Mod Via the qdomyos-zwift wiki.  I'm currently using QZ with my IC8 for my wife to take peloton classes and to improve the integration with Zwift for me, and it works great.  Even after a few week I'm a bit gutted I did not look for a bike with auto resistance, but this mod looks fantastic as I love everything else a out the IC8.  I've built a few DIY projects so I'm okay with soldering etc.

I've already order the PCB and most of the parts from Ali Express, and happened to acquire an Ender 3 printer from a friend a few weeks back, so this will be a great project to learn 3D printing with.  I'm not sure what I am going to do about printing the shifter yet as that seems a little more complicated materials wise but since the stuff from AliExpress will possibly take months to arrive I have plenty of time to keep reading up.

I love the look of @eMadman's IC4 model with the improved mount to the bike.  When reading that page I noticed you said you adapted V2 case slightly to fit your step motor.  Since I'm new to 3d printing I'm hoping to minimise having to adapt things initially so I was wondering if you could let me know the exact size of your motor so I can find the same.

Also in general I need to keep the bike wife friendly so can you confirm you IC8 mod is easy to remove for her to continue to take peloton classes, it seems so from the hook on your mount. Although it also seems that the QZ app will allow her to use auto resistance in peloton so might not even need removing 👍

Sorry for long initial post 😁

Ash

**Comment by doudar** [comment]
@Mightydob , Welcome!
I’ll let @eMadman back me up and verify this but I believe he changed the width of the stepper wire pass through gap to accommodate a wider connector that his motor had. So the original length motor will still work.

**Comment by eMadman** [comment, marked as answer]
@Mightydob!  I'm glad you're taking on this project!  It's always great to see someone printing and building one themselves!

>  I'm not sure what I am going to do about printing the shifter yet as that seems a little more complicated materials wise but since the stuff from AliExpress will possibly take months to arrive I have plenty of time to keep reading up.

Printing TPU is a bit of a pain in the butt.  Having a direct drive extruder helps a lot but isn't mandatory.  You can frequently get away with printing the material if you go very slowly and disable retractions.  I'm not familiar with the extruder on the Ender to be able to say it's possible or not.  I think with standard TPU, the material should be just stiff enough to be okay.  

> I love the look of @eMadman's IC4 model with the improved mount to the bike. When reading that page I noticed you said you adapted V2 case slightly to fit your step motor. Since I'm new to 3d printing I'm hoping to minimise having to adapt things initially so I was wondering if you could let me know the exact size of your motor so I can find the same.

You won't have to make any changes regarding the motor.  I had an odd flange around the stepper motor wire connector that I made room for.  Any standard 38mm long Nema 17 motor will do the job.  I [ordered these most recently](https://www.amazon.ca/gp/product/B07THK76QQ/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1)

These dimensions come off a motor I've used with success recently.  42x42x38mm:
![61XfdYgUhjL _SL1500_](https://user-images.githubusercontent.com/24726844/150349995-6ae77904-eef7-4b03-bcf6-c8399c43d278.jpg)

> Also in general I need to keep the bike wife friendly so can you confirm you IC8 mod is easy to remove for her to continue to take peloton classes, it seems so from the hook on your mount.

The hook is there for easy removal.  I've got the tolerances on the design pretty snug so it might need a little bit of sanding to make for easier easy for 1-handed removal.  The mount that fixes to the head tube of the bike is using a velcro cable tie for easy adjustment.  This came as a result of user feedback... the guy and his partner had a huge height difference between them and he needed to move the bike mount up and down between the two of them.  

> Although it also seems that the QZ app will allow her to use auto resistance in peloton so might not even need removing 👍

Short answer:  Not working yet.  
Long answer:  It's getting close!  One of my more recent rides showed a lot of promise in how it felt.  Roberto at QZ will be receiving a unit for his own testing soon.  With that, I think it will be fixed much more quickly.  There is an issue in the QZ bug tracker and a couple of us have been helping test and generate logs for it.   [Check it out here](https://github.com/cagnulein/qdomyos-zwift/issues/500)

Until the integration is ready for a larger audience, I have instructions hiding out in a [google doc here.](https://docs.google.com/document/d/1jG0Lkv-nrNkndlw4ikXSDv9OeyQTjr0w_uEBPLya-vg/edit#).  Once its ready for daily use, I'll merge this content into the SS2k Wiki.

**Comment by Mightydob** [comment]
Brilliant thanks both, really excited to get started on this from a geeky point of view 😁.

Great about the motor will get one ordered.

Re TPU I did have a quick read up about the Ender 3 and it said it was possible if the TPU isn't too flexible, such as one with a Shore Hardness of 98A, I've no idea if that's compatible with the print for the strap.

Had a quick read of the integration with QZ and sounds like things are progressing, it will be a while before I'm up and running I'm sure, so will keep an eye on it.  I'm happy to remove SS2k and install whenever I use the bike if it's not fixed by then 👍

Thanks for quick replies 

Ash

**Reply by doudar** [reply]
You shouldn't have any issues with the ender 3. Just go slow with the TPU. I had great results my first few times using Chep's TPU profiles from: https://www.chepclub.com/cura-profiles.html

**Reply by eMadman** [reply]
> Re TPU I did have a quick read up about the Ender 3 and it said it was possible if the TPU isn't too flexible, such as one with a Shore Hardness of 98A, I've no idea if that's compatible with the print for the strap.

I'm using TPU from Sainsmart... it's a roll I've had for at least 4 years so their formulation may have changed, but tpu should be around that 95A shore hardness.  I don't think 98 is going to be problematic.  Let's assume it is - i think what you'll experience is inflexibility in the strap for the shifter closest to where the button/wire pod is.  Rather than reprinting, if you run into this, I think an easy fix can be found using a shim to increase the diameter of the handlebar.  

Longer-term, I want to make a variation of the shifter design that uses a lot less TPU... potentially a TPU button pod and some sort of plastic design to hold it in place using  a clamp mount.

**Reply by Mightydob** [reply]
👍 think I need to start messing with the printer over the next few weeks, got plenty if different little projects I've been saving up till I got one.  I will take a look at those profiles, I played with Cura the other night, so profiles make sense I think.  Sounds promising about the strap design plans I can see how making it more like how other bike accessories connect e.g. bells, lights etc. Make sense as it's a tried and tested strong method.

**Reply by doudar** [reply]
If you're new to 3d printing, most likely your biggest issue will be bed adhesion (especially if you use PETG or ABS.) If you have any issues with bed adhesion, I strongly recommend you get a magnetic PEI spring steel sheet for on your printer. It's an easy upgrade and will save you hours of headaches and wasted prints. Just make sure to let it cool down completely before you remove a print. You can speed that up a bit with some 99% isopropyl. Just pour some around the base of your print when it's finished.

**Reply by Mightydob** [reply]
Great tips thanks, printer was super cheap so don't mind spending a little on upgrades
